models/repo.go

```diff
@@ -1242,14 +1242,16 @@ func Repositories(opts *SearchRepoOptions) (_ RepositoryList, err error) {
 		Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).
 		OrderBy(opts.OrderBy).
 		Find(&repos); err != nil {
-		return nil, fmt.Errorf("Repo: %v", err)
+		return nil, 0, fmt.Errorf("Repo: %v", err)
 	}
 
 	if err = repos.loadAttributes(x); err != nil {
-		return nil, fmt.Errorf("LoadAttributes: %v", err)
+		return nil, 0, fmt.Errorf("LoadAttributes: %v", err)
 	}
 
-	return repos, nil
+	count = countRepositories(-1, opts.Private)
```

Comment:
Here, we check `opts.Private` to see if we should include private repos, but when populating the `repos` slice, we always include private repos regardless of `opts.Private`